core/commands/list_cloud_instances_command.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'xdg'
+require 'json'
+require 'yaml'
+require 'tty-table'
+
+require_relative 'base_command'
+require_relative '../services/aws_service'
+require_relative '../services/gcp_service'
+
+# Command shows list all active instances on Cloud Providers
+class ListCloudInstancesCommand < BaseCommand
+  HIDDEN_GCP_INSTANCES_BY_USER = 'mdbci/hidden-gcp-instances.yaml'
+  HIDDEN_GCP_INSTANCES_BY_DEFAULT = '../../config/hidden-gcp-instances.yaml'
+
+  def self.synopsis
+    'Show list all active instances on Cloud Providers'
+  end
+
+  def show_help
+    info = <<-HELP
+List cloud instances command shows a list of active machines on GCP and AWS providers and the time they were created.
+    HELP
+    @ui.info(info)
+  end
+
+  def execute
+    if @env.show_help
+      show_help
+      return SUCCESS_RESULT
+    end
+    show_list
+    SUCCESS_RESULT
+  end
+
+  def show_list
+    if @env.aws_service.configured?
+      show_aws_list
+    else
+      @ui.info('AWS-service is not configured. AWS-list is not available.')
+    end
+    if @env.gcp_service.configured?
+      show_gcp_list
+    else
+      @ui.info('GCP-service is not configured. GCP-list is not available.')
+    end
+  end
+
+  def show_aws_list
+    all_instance = @env.aws_service.instances_list.sort do |first, second|
+      first[:launch_time] <=> second[:launch_time]
+    end
+    @ui.info('List all active instances on AWS:')
+    @ui.out(JSON.generate({ aws: all_instance }))
```

Comment:
Please introduce the `--json` flag and use it to differentiate between machine readable output and human-readable output.